account_statement_import_txt_xlsx/models/account_statement_import_sheet_parser.py

```diff
@@ -0,0 +1,389 @@
+                line["notes"] = notes
+            if reference is not None:
+                line["reference"] = reference
+            if partner_name is not None:
+                line["partner_name"] = partner_name
+            if bank_name is not None:
+                line["bank_name"] = bank_name
+            if bank_account is not None:
+                line["bank_account"] = bank_account
+            lines.append(line)
+        return lines
+
+    @api.model
+    def _convert_line_to_transactions(self, line):  # noqa: C901
+        """Hook for extension"""
+        timestamp = line["timestamp"]
+        amount = line["amount"]
+        currency = line["currency"]
+        original_amount = line["original_amount"]
+        original_currency = line["original_currency"]
+        transaction_id = line.get("transaction_id")
+        description = line.get("description")
+        notes = line.get("notes")
+        reference = line.get("reference")
+        partner_name = line.get("partner_name")
+        bank_name = line.get("bank_name")
+        bank_account = line.get("bank_account")
+
+        transaction = {
+            "date": timestamp,
+            "amount": str(amount),
+        }
+
+        if original_currency == currency:
+            original_currency = None
+            if not amount:
+                amount = original_amount
+            original_amount = "0.0"
+
+        if original_currency:
+            original_currency = self.env["res.currency"].search(
+                [("name", "=", original_currency)],
+                limit=1,
+            )
+            if original_currency:
+                transaction["foreign_currency_id"] = original_currency.id
+            if original_amount:
+                transaction["amount_currency"] = str(original_amount)
+
+        if currency:
+            currency = self.env["res.currency"].search(
+                [("name", "=", currency)],
+                limit=1,
+            )
+            if currency:
+                transaction["currency_id"] = currency.id
+
+        if transaction_id:
+            transaction["unique_import_id"] = "{}-{}".format(
+                transaction_id,
+                int(timestamp.timestamp()),
+            )
+
+        transaction["name"] = description or _("N/A")
+        if reference:
+            transaction["ref"] = reference
+
+        note = ""
+        if bank_name:
+            note += _("Bank: %s; ") % (bank_name,)
+        if bank_account:
+            note += _("Account: %s; ") % (bank_account,)
+        if transaction_id:
+            note += _("Transaction ID: %s; ") % (transaction_id,)
+        if note and notes:
+            note = "{}\n{}".format(note, note.strip())
+        elif note:
+            note = note.strip()
+        transaction["payment_ref"] = (
+            note
+            or reference
+            or description
+            or transaction.get("unique_import_id")
+            or "/"
+        )
```

Comment:
```suggestion
        if note:
            transaction["narration"] = note
```
I think now in v14 `narration` field is the substitute for `note` field, so I think we should only change `note` to `narration` and leave everything as it was before, what do you think?

Reply:
I maintain that we should try and populate `payment_ref` to the best of our possibilities. Contrary to other modules, we don't have control over the data we are provided. `description` may, or may not, be populated (that depends on the map).

Reply:
> I think now in v14 `narration` field is the substitute for `note` field, so I think we should only change `note` to `narration` and leave everything as it was before, what do you think?

You are right, fixed.
Thanks for your review!